.github/workflows/container.yml

```diff
@@ -0,0 +1,57 @@
+name: Create and publish a container image
+
+on:
+  push:
+    # Publish main branch as `beta` image
+    branches:
+      - main
+      - master
+
+    # Publish tags as versioned release and `latest` image
+    tags:
+      - '*'
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Log in to registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
```

Comment:
suggestion:

Use the existing [action for logging in](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages)

Reply:
"A little copying is better than a little dependency."

That action is for unknown reasons pinned to specific version in the documentation you linked, and using third party action instead of one-liner is IMHO harder to read than this, which is widely used.